### PR TITLE
[Indexer] General improvements and fixes

### PR DESCRIPTION
**Pull request by sideninja**
This PR contains multiple unrelated improvements and fixes found during the additional testing.

______

For contributor use:

- [ ] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [standards mentioned here](https://github.com/onflow/flow-nft/blob/master/CONTRIBUTING.md#styleguides).
- [ ] Updated relevant documentation 
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

File: api/server.go [hunk 1]
```diff
@@ -203,18 +206,24 @@ func (h *httpServer) disableRPC() bool {
...
-		if checkPath(r, "/ws") {
+		if checkPath(r, "") {
 			ws.ServeHTTP(w, r)
 			return
 		}
 	}
 
 	// If JSON-RPC over HTTP is enabled, try to serve the request
-	rpc := h.httpHandler
+	rpc := recoverHandler(h.log, h.httpHandler)
```
Comment (m-Peter) [inline api/server.go:224]: This should also wrap the CORS handler we introduced in main.

File: services/ingestion/subscriber.go [hunk 2]
```diff
@@ -33,5 +34,10 @@ func (r *RPCSubscriber) Subscribe(ctx context.Context, height uint64) (<-chan fl
 		},
 	}
 
-	return r.client.SubscribeEventsByBlockHeight(ctx, height, filter)
+	evs, errs, err := r.client.SubscribeEventsByBlockHeight(ctx, height, filter)
```
Comment (m-Peter) [inline services/ingestion/subscriber.go:37]:
```suggestion
	evs, errs, err := r.client.SubscribeEventsByBlockHeight(ctx, height, filter, grpc.WithHeartbeatInterval(1))
```

We actually need to use the above variant, but this is only available for this client:
```go
client, err := grpc.NewBaseClient(
	config.AccessURL,
	goGrpc.WithTransportCredentials(insecure.NewCredentials()),
)
```
If it's too much trouble to add it now, we should add a TODO and tackle it later.

**Review by m-Peter — APPROVED** [review]
Looks good overall :clap: 
Only a comment about the grpc client we should use from `flow-go-sdk`. Not a blocker for this PR though, can be fixed in a different PR.

File: services/ingestion/subscriber.go (hunk 2 above)
Comment (sideninja) [reply services/ingestion/subscriber.go:42]: I believe the default heartbeat is 100, which is working well for now, but I will add todo if we should revisit it.

**Pull request merged**